### PR TITLE
UCT/IB/MLX5: Fasten DC support check

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

File: src/uct/ib/mlx5/ib_mlx5_dv.c [hunk 1]
```diff
@@ -50,24 +54,63 @@ static ucs_status_t uct_ib_mlx5_device_init(uct_ib_device_t *dev)
         goto err_cq;
     }
 
+    srq_attr.attr.max_sge   = 1;
+    srq_attr.attr.max_wr    = 1;
+    srq = ibv_create_srq(pd, &srq_attr);
+    if (srq == NULL) {
+        ucs_error("ibv_create_cq() failed: %m");
```
Comment (yosefe) [inline src/uct/ib/mlx5/ib_mlx5_dv.c]: ibv_create_srq ?

File: src/uct/ib/mlx5/ib_mlx5_dv.c [hunk 2]
```diff
@@ -50,24 +54,63 @@ static ucs_status_t uct_ib_mlx5_device_init(uct_ib_device_t *dev)
...
+                                   IBV_QP_ACCESS_FLAGS);
+    if (ret != 0) {
+        goto err;
+    }
+
+    attr.qp_state                  = IBV_QPS_RTR;
+    attr.path_mtu = IBV_MTU_256;
+    attr.ah_attr.port_num = 1;
+
+    ret = ibv_modify_qp(qp, &attr, IBV_QP_STATE |
```
Comment (yosefe) [inline src/uct/ib/mlx5/ib_mlx5_dv.c]: maybe extract new function just for testing if DC works?

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9120/ for details (Mellanox internal link).

File: src/uct/ib/mlx5/ib_mlx5_dv.c [hunk 3]
```diff
@@ -50,24 +54,63 @@ static ucs_status_t uct_ib_mlx5_device_init(uct_ib_device_t *dev)
...
+                           IBV_ACCESS_REMOTE_ATOMIC;
+    ret = ibv_modify_qp(qp, &attr, IBV_QP_STATE |
+                                   IBV_QP_PKEY_INDEX |
+                                   IBV_QP_PORT |
+                                   IBV_QP_ACCESS_FLAGS);
+    if (ret != 0) {
+        goto err;
+    }
+
+    attr.qp_state                  = IBV_QPS_RTR;
```
Comment (brminich) [inline src/uct/ib/mlx5/ib_mlx5_dv.c]: indentation

File: src/uct/ib/mlx5/ib_mlx5_dv.c [hunk 4]
```diff
@@ -50,24 +54,63 @@ static ucs_status_t uct_ib_mlx5_device_init(uct_ib_device_t *dev)
...
 
-    /* create DCI qp successful means DC is supported */
+    /* create DCT qp successful means DC is supported */
     qp = mlx5dv_create_qp(ctx, &qp_attr, &dv_attr);
-    if (qp) {
-        ibv_destroy_qp(qp);
+    if (qp == NULL) {
+        goto err_qp;
+    }
+
```
Comment (brminich) [inline src/uct/ib/mlx5/ib_mlx5_dv.c]: is not creating a qp enough to check DC support?
Why is it required to move it to RTR state?

File: src/uct/ib/mlx5/ib_mlx5_dv.c (hunk 2 above)
Comment (Artemy-Mellanox) [reply src/uct/ib/mlx5/ib_mlx5_dv.c]: This would entire old function.

File: src/uct/ib/mlx5/ib_mlx5_dv.c (hunk 4 above)
Comment (Artemy-Mellanox) [reply src/uct/ib/mlx5/ib_mlx5_dv.c]: Move to RTR state triggers in kernel actual CREATE_DCT

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9199/ for details (Mellanox internal link).

**Pull request merged**